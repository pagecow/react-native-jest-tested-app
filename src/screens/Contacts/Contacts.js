import React, { useState, useEffect } from 'react';
import { View, ActivityIndicator } from 'react-native';
import { useIsFocused } from '@react-navigation/native';
import { ScrollView } from 'react-native-gesture-handler';
import styles from './Contacts.style';

import ContactsCont from './components/ContactsCont/ContactsCont';

export default function Contacts({ navigation, allContactsInfo, favContactsInfo, contactsInfo, setChosenContactInfo }) {
  const [favContactsInfoUpdated, setFavContactsInfoUpdated] = useState([]);
  const [contactsInfoUpdated, setContactsInfoUpdated] = useState([]);

  const isFocused = useIsFocused();

  useEffect(() => {
    if(allContactsInfo){
      setFavContactsInfoUpdated(allContactsInfo.filter(e => e.isFavorite === true));
      setContactsInfoUpdated(allContactsInfo.filter(e => e.isFavorite === false));
    }
  }, [isFocused])
  
  if(!contactsInfo?.length && !favContactsInfo?.length) {
    return (
    <ActivityIndicator 
        style={styles.activityIndicator}
        size="large"
        color="#6c94ad"
    />
    );
  } else if(!favContactsInfoUpdated?.length && !contactsInfoUpdated?.length) {
    return (
      <View style={styles.screen}>
        <ScrollView>
          <ContactsCont 
            fav={true} 
            info={favContactsInfo} 
            setChosenContactInfo={setChosenContactInfo} 
            navigation={navigation}
          />
          <ContactsCont 
            fav={false} 
            info={contactsInfo} 
            setChosenContactInfo={setChosenContactInfo} 
            navigation={navigation}
          />        
        </ScrollView>
      </View>
    );
  } else {
    return (
      <View style={styles.screen}>
        <ScrollView>
          <ContactsCont 
            fav={true} 
            info={favContactsInfoUpdated} 
            setChosenContactInfo={setChosenContactInfo} 
            navigation={navigation}
          />
          <ContactsCont 
            fav={false} 
            info={contactsInfoUpdated} 
            setChosenContactInfo={setChosenContactInfo} 
            navigation={navigation}
          />        
        </ScrollView>
      </View>
    );
  }
};