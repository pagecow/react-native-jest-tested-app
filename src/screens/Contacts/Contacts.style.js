import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    activityIndicator: {
        position: "absolute",
        left: 0,
        right: 0,
        bottom: 0,
        top: 0,
    },
    screen: {
        flex: 1, 
        backgroundColor: "#fff",
    }
});