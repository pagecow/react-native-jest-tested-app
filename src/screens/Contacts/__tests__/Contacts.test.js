import React from 'react';
import renderer from 'react-test-renderer';

import Contacts from '../Contacts';
jest.mock('@react-navigation/native');

describe('<Contacts />', () => {
    it('renders without crashing', () => {
        const tree = renderer.create(<Contacts />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
