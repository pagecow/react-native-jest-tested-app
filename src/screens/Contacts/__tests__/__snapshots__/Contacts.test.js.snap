// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contacts /> renders without crashing 1`] = `
<ActivityIndicator
  animating={true}
  color="#6c94ad"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
